documents/description.md

```diff
@@ -2,8 +2,105 @@
+to novices while also supporting a plethora of features neccessary for more interesting challenges. This domain is useful since it can
+be both entertaining and educational!
+
+###### Who will benefit from this language?
+As mentioned above, although they will get different things out of it, this language should ideally benefit both novice and
+experienced programmers. That is, a novice programmer should get an interactive educational and fun experience while a experienced 
+programmer should get an interactive challenge with regards to AI and naviagation.
+
+###### Are there any other DSLs for this domain? If so, what are they, and how might they influence your language design and implementation?
+There is nothing I have discovered that is exactly like this idea. That being said, the idea for this project was inspired by
+_SCM Draft 2_, _StarForge_, the _Warcraft III Map Editor_, and _Game Maker_. In the case of the first three inspirations, these
+are all map editors to create custom scenarios in real time strategy (RTS) games. While coding is not part of the actual game,
+here these map editors allow you to define a game within either _Starcraft_ or _Warcraft III_'s engine. Game Maker on the other hand
+is a drag and drop and scripting DSL made for enabling easy development of 2D and simple 3D games. Although this also does not make
+coding "part of the game", the simplified language design is an inspiration for the API-like interface that will be used within this 
+game.
+
+This idea is unique since the game itself is to code. It is not just making a game with code.
+
+
+
 ## Language design
 
+###### If you had to capture your DSL's design in one sentence, what would it be?
+Providing coding feedback and error checking within a game enviornment.
+
+###### What constitutes a program in your language?
+A program will look very much like a simple C-like program. There will be a "main" function that the game hooks into and uses to 
+define the behavior of various spaceships within the game. The user may also define helper functions and global variables to aid in
+defining behavior.
+
+###### What happens when a program runs?
+If the code works, it will define the behavior of a spaceship. Examples of this behavior may be to fly at various speeds, shoot at 
+enemies, navigate obstacles, etc.
+
+If a program has either a syntax or runtime error, however, the game should _not_ crash. Rather the game should provide feedback
+to the user. For instance, if a user forgets a semicolon, a character in the game should tell the player "Hey! I think you forgot a
+semicolon here!"
+
+###### What kinds of input might a program take, and what kinds of output might it produce?
+The input of a program would be calls to the spaceship's API. This will allow the code to access the spaceships environment and make
+decisions based on its surroundings. Examples of these might be range finders, hull status, and receiving signals from a flagship.
+
```

Comment:
I'm confused about the input. How do "calls to the spaceship's API" work as inputs to a program in your DSL?